accepted/00052-new-package-predownloading.md

```diff
@@ -0,0 +1,85 @@
+ - we want users to be able to change the download schedule without restarting Tomcat and Taskomatic (currently the case as parameters are in `rhn.conf`)
+
+# Detailed design
+
+## User interface
+
+### GUI
+
+Whenever a package installation or patch application is scheduled, users will need a new checkbox to schedule a "download only" Action:
+
+![GUI diff animation](images/00052-gui-diff.gif)
+
+The checkbox will not appear for traditionally managed systems.
+
+This will need changes in the following pages:
+
+- Single systems:
+  - patch application: https://server.tf.local/rhn/systems/details/ErrataList.do?sid=1000010000&
+  - package installation: https://server.tf.local/rhn/systems/details/packages/InstallPackages.do?sid=1000010000&
+  - package upgrade: https://server.tf.local/rhn/systems/details/packages/UpgradableList.do?sid=1000010000&
+
+- SSM:
+  - patch application: https://server.tf.local/rhn/systems/ssm/ListErrata.do
+  - package installation: https://server.tf.local/rhn/ssm/PackageInstall.do
+  - package upgrade: https://server.tf.local/rhn/ssm/PackageUpgrade.do
+
+*Links above assume `server.tf.local` as the server location and `1000010000` as a minion ID.*
+
+### API
+
+The following calls will need a new optional `downloadOnly` boolean flag:
+- `actionchain.addErrataUpdate`
+- `actionchain.addPackageInstall`
+- `actionchain.addPackageUpgrade`
+- `actionchain.schedulePackageInstall`
+- `actionchain.schedulePackageInstallByNevra`
+- `system.schedulePackageInstall`
+- `system.schedulePackageInstallByNevra`
+- `system.scheduleApplyErrata`
+- `systemgroup.scheduleApplyErrataToActive`
+
+## Implementation
+- new checkbox item will be added to current JSPs/React components
+  - checkbox has to be shown only if applicable systems are all minions
+- existing Java handling code will receive the new flag, pass it down to Salt calling code in order to apply
+  - [pkgdownload.sls](https://github.com/uyuni-project/uyuni/blob/c8ffe6b9425392f5235864ad070646bb8ebc2ecb/susemanager-utils/susemanager-sls/salt/packages/pkgdownload.sls) instead of [pkginstall.sls](https://github.com/uyuni-project/uyuni/blob/c8ffe6b9425392f5235864ad070646bb8ebc2ecb/susemanager-utils/susemanager-sls/salt/packages/pkginstall.sls) if set
+  - [patchdownload.sls](https://github.com/uyuni-project/uyuni/blob/c8ffe6b9425392f5235864ad070646bb8ebc2ecb/susemanager-utils/susemanager-sls/salt/packages/patchdownload.sls) instead of [patchinstall.sls](https://github.com/uyuni-project/uyuni/blob/c8ffe6b9425392f5235864ad070646bb8ebc2ecb/susemanager-utils/susemanager-sls/salt/packages/patchinstall.sls) if set
+  - alternatively, those files might be unified and use `pkg.downloaded` instead of `pkg.installed` conditionally (resp. `pkg.patch_downloaded` instead of `pkg.patch_installed`)
+- existing configuration options are to be deprecated (in SUSE Manager supported versions) or removed
+  - existing code handling the configuration options can be removed from Uyuni
+
+# Drawbacks
```

Comment:
Another drawback to be added: from the user point of view, we will have two different options in two different locations to enable content staging for traditional and minions.

Reply:
That also used to be the case before - the difference is now the two mechanisms are also conceptually different. Clarified.